packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -860,6 +910,9 @@ class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigation
           alwaysShowMiddle: widget.alwaysShowMiddle && widget.middle != null,
           stretchConfiguration: widget.stretch ? OverScrollHeaderStretchConfiguration() : null,
           enableBackgroundFilterBlur: widget.enableBackgroundFilterBlur,
+          bottom: widget.bottom ?? const SizedBox.shrink(),
+          bottomMode: widget.bottomMode,
+          bottomHeight: widget.bottom != null && widget.bottomMode != NavigationBarBottomMode.none ? widget.bottom!.preferredSize.height : 0.0,
```

Comment:
Should we have an assert that complains if we are given a bottom widget and a `bottomMode` of none?

Reply:
I don't know to be honest. In that case I can imagine we don't even need a `.none` mode. We can just default to `.automatic`, then not care about the bottom unless one is supplied. Or we can default to `.always` and rename `.automatic` to `.hide` or something to avoid confusion.

Reply:
My vote is remove `.none` and default to `.automatic`. I believe that's how the default is in SwiftUI.

I don't believe we should use the word "hide" because that may infer that the widget stays hidden.